### PR TITLE
Tests for commands of the SRAweb module

### PR DESCRIPTION
**Pull request by dibyaaaaax**

### CODE REVIEW

**Comment by dibyaaaaax** [comment]
@saketkc could you please review the tests I have written? I will complete the rest of them once I am sure I'm going in the right direction.

**Comment by saketkc** [comment]
Looks good! Can you rebase with master? You might get conflicts, but they should be easy to resolve.

**Comment by dibyaaaaax** [comment]
I don't understand why Travis fails while running the tests. I didn't get any error while running them on my machine. @saketkc could you please help me out?

**Comment by saketkc** [comment]
Thanks @dibyaaaaax! This looks like a good solid test set. Sorry for the delay in getting back to you, I was traveling.

The failure on Python 3.7 appears to be a random failure. I have restarted the build. Let's see if it goes through. If not, I will have to dig deeper.

**Comment by dibyaaaaax** [comment]
Thank you @saketkc! Hope it works :smiley:

**Comment by saketkc** [comment]
Something still wrong with Py3.7. I will have to dig deeper.

File: tests/test_sraweb.py [hunk 1]
```diff
@@ -14,3 +14,138 @@ def sraweb_connection():
 def test_sra_metadata(sraweb_connection):
     df = sraweb_connection.sra_metadata("SRP016501")
     assert df.shape[0] == 134
+
+
+def test_tissue_column(sraweb_connection):
```
Comment (saketkc) [inline tests/test_sraweb.py]: Can you add one line description to each method such as """Test if tissue column exists""".

File: tests/test_sraweb.py [hunk 2]
```diff
@@ -14,3 +14,138 @@ def sraweb_connection():
...
+
+
+def test_gse_to_gsm(sraweb_connection):
+    df = sraweb_connection.gse_to_gsm("GSE56924", detailed = False)
+    assert df.shape[0] == 96
+
+
+def test_gse_to_srp(sraweb_connection):
+    df = sraweb_connection.gse_to_srp("GSE63858")
+    print(df)
```
Comment (saketkc) [inline tests/test_sraweb.py]: Please remove this.

**Comment by saketkc** [comment]
HI @dibyaaaaax, thanks for your continued patience. We are almost there. The error is because of tests exceeding the API rate limits. A simple fix wouldbe to add `time.sleep(1)` in the fixture:

```
@pytest.fixture(scope="module")
def sraweb_connection():
    db = SRAweb()
    time.sleep(1)
    return db
```

**Comment by dibyaaaaax** [comment]
@saketkc the API rate limit exceeds even with the sleep time.

**Comment by saketkc** [comment]
Can you give me edit permissions for this PR @dibyaaaaax? Thanks! 
https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by dibyaaaaax** [comment]
@saketkc you already have the edit permission.

**Comment by saketkc** [comment]
@dibyaaaaax, thanks a lot for this major effort! We are good to go to the master.

**Comment by saketkc** [comment]
Are you okay with me adding your Name to [AUTHORS](https://github.com/saketkc/pysradb/blob/master/AUTHORS.rst):

Contributors
----------------

- Dibya Gautam <https://github.com/dibyaaaaax>

**Comment by dibyaaaaax** [comment]
yeah sure, thanks a lot @saketkc!

**Pull request merged**